galadriel/tools/web3/market_data/dextools.py

```diff
@@ -0,0 +1,639 @@
+
+    FREE = "free"
+    TRIAL = "trial"
+    STANDARD = "standard"
+    ADVANCED = "advanced"
+    PRO = "pro"
+    PARTNER = "partner"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class DexToolsBase(Tool, ABC):
+    """Base class for DEXTools API tools."""
+
+    def __init__(self, plan: DexToolsPlan = DexToolsPlan.TRIAL):
+        super().__init__()
+        if not DEXTOOLS_API_KEY:
+            raise ValueError("DEXTOOLS_API_KEY environment variable is not set")
+        self.client = DextoolsAPIV2(DEXTOOLS_API_KEY, plan=str(plan))
+        self.plan = plan
+
+
+class GetBlockchainInfoTool(DexToolsBase):
+    """Tool for fetching blockchain information from DEXTools."""
+
+    name = "get_dextools_blockchain"
+    description = "Get blockchain information from DEXTools"
+    inputs = {
+        "chain": {
+            "type": "string",
+            "description": "The blockchain identifier (e.g., 'ether', 'bsc', 'polygon')",
+        }
+    }
+    output_type = "string"
+
+    def forward(self, chain: str) -> str:
+        """Fetch blockchain information.
+
+        Args:
+            chain: The blockchain identifier
+
+        Returns:
+            str: Blockchain information as JSON string
+        """
+        try:
+            response = self.client.get_blockchain(chain)
+            return json.dumps(response, indent=2)
+        except Exception as e:
+            logger.error(f"Failed to fetch blockchain info: {e}")
+            return json.dumps({"error": str(e)})
+
+
+class GetBlockchainsTool(DexToolsBase):
+    """Tool for fetching all blockchains from DEXTools."""
+
+    name = "get_dextools_blockchains"
+    description = "Get all blockchains from DEXTools with optional sorting"
+    inputs = {
+        "sort": {
+            "type": "string",
+            "description": "Sort field (e.g., 'name')",
+            "nullable": True,
+        },
+        "order": {
+            "type": "string",
+            "description": "Sort order ('asc' or 'desc')",
+            "nullable": True,
+        },
+    }
+    output_type = "string"
+
+    def forward(
+        self,
+        sort: Optional[str] = None,
+        order: Optional[str] = None,
+    ) -> str:
+        """Fetch all blockchains with optional sorting.
+
+        Args:
+            sort: Sort field
+            order: Sort order ('asc' or 'desc')
+
+        Returns:
+            str: Blockchains information as JSON string
+        """
+        try:
+            response = self.client.get_blockchains(sort=sort, order=order)
+            return json.dumps(response, indent=2)
+        except Exception as e:
+            logger.error(f"Failed to fetch blockchains: {e}")
+            return json.dumps({"error": str(e)})
+
+
+class GetDexFactoryInfoTool(DexToolsBase):
+    """Tool for fetching DEX factory information from DEXTools."""
+
+    name = "get_dextools_dex_factory"
+    description = "Get DEX factory information from DEXTools"
```

Comment:
Similar to get blockchain info - shouldn't it explain what DEX factory is and in which cases it may be useful?